File: Data Structures & Algorithms/concatenation-of-array/submission-0.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int[] ans = new int[2 * nums.length];
        int j = 0;
        for (int i = 0; i < 2;i++){
            for (int num : nums){
                ans[j++] = num;
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prod = 1
        zero_cnt = 0
        for n in nums:
            if n:
                prod *= n
            else:
                zero_cnt +=1
        if zero_cnt > 1:
            return [0] * len(nums)
        res = [0] * len(nums)
        for i, c in enumerate(nums):
            if zero_cnt == 1:
                if c != 0:
                    res[i] = 0
                else:
                    res[i] = prod
            else:
                res[i] = prod // c
        return res